packages/core/src/generateKernel.js

```diff
@@ -38,19 +38,23 @@ function generateKernel (opts = {}) {
   output = replaceTemplateRequire(output, 'ses', sesSrc)
   output = stringReplace(output, '__createKernelCore__', kernelCode)
   output = stringReplace(output, '__lavamoatDebugOptions__', JSON.stringify({debugMode: !!opts.debugMode}))
-  if (opts.hasOwnProperty('scuttleGlobalThis')) {
+  if (opts?.hasOwnProperty('scuttleGlobalThis')) {
     // scuttleGlobalThis config placeholder should be set only if ordered so explicitly.
     // if not, should be left as is to be replaced by a later processor (e.g. LavaPack).
-    const {scuttleGlobalThis, scuttleGlobalThisExceptions} = opts
-    if (scuttleGlobalThisExceptions) {
+    const scuttleGlobalThis = opts.scuttleGlobalThis
+    if (opts.scuttleGlobalThisExceptions) {
+      console.warn('Lavamoat - "scuttleGlobalThisExceptions" is deprecated. Use "scuttleGlobalThis.exceptions" instead.')
+    }
+    const exceptions = scuttleGlobalThis?.exceptions || opts.scuttleGlobalThisExceptions
+    scuttleGlobalThis.exceptions = exceptions
```

Comment:
```suggestion
    if (opts.scuttleGlobalThisExceptions) {
      console.warn('Lavamoat - "scuttleGlobalThisExceptions" is deprecated. Use "scuttleGlobalThis.exceptions" instead.')
    }
    const scuttleGlobalThis = {
      ...opts.scuttleGlobalThis,
      exceptions: opts.scuttleGlobalThis?.exceptions || opts.scuttleGlobalThisExceptions,
    }
    const exceptions = scuttleGlobalThis.exceptions
```
This way we leave the `opts` object unmodified.
I also personally find it more readable to assign all the options fields ni one statement like this, but that might be subjective.

Reply:
706199a